### PR TITLE
20200803 checker

### PR DESCRIPTION
**Pull request by changlong-liu**

### CODE REVIEW

File: src/utils/helper.ts [hunk 1]
```diff
@@ -315,4 +315,4 @@ export function parseResourceId(mpath: string): Map<string, string> {
         ret.set("last_child_num", "" + count);
     }
     return ret;
-}
```
Comment (LianwMS) [inline src/utils/helper.ts:318]: undo change since no actually changes

File: src/plugins/azgenerator/TemplateAzureCliTestScenario.ts [hunk 2]
```diff
@@ -48,13 +47,13 @@ export function GenerateAzureCliTestScenario(model: CodeModelAz): string[] {
...
-            model.FindExampleById(exampleId, commandParams);
-        }
-    }
+    // // go through the examples to recognize resources
+    // for (var ci = 0; ci < config.length; ci++) {
+    //     let exampleId: string = config[ci].name;
+    //     if (exampleId) {
+    //         model.FindExampleById(exampleId, commandParams, []);
+    //     }
+    // }
```
Comment (LianwMS) [inline src/plugins/azgenerator/TemplateAzureCliTestScenario.ts:56]: Remove if no needed

File: src/plugins/azgenerator/ScenarioTool.ts [hunk 3]
```diff
@@ -201,41 +200,169 @@ class ResourceClass {
     }
 
 }
+
+export enum ObjectStatus {
+    NA,
```
Comment (LianwMS) [inline src/plugins/azgenerator/ScenarioTool.ts:205]: >NA [](start = 4, length = 2)

None?

File: src/plugins/azgenerator/TemplateAzureCliTestScenario.ts [hunk 4]
```diff
@@ -101,7 +103,21 @@ export function GenerateAzureCliTestScenario(model: CodeModelAz): string[] {
                         let postfix: string = (idx < exampleCmd.length - 1) ? " '" : "',";
                         ToMultiLine(prefix + exampleCmd[idx] + postfix, steps);
                     }
-                    steps.push("    " + disabled + "         checks=[])");
+                    let checks =  model.GetExampleChecks(examples[exampleIdx++]);
+                    if (checks.length>0) {
```
Comment (LianwMS) [inline src/plugins/azgenerator/TemplateAzureCliTestScenario.ts:107]: >if (checks.length>0 [](start = 20, length = 19)

format the code here

**Comment by qiaozha** [comment]
Hold on this PR until we confirmed with CLI guys.

**Comment by qiaozha** [comment]
@changlong-liu  Cam we add the not-generate check option in the e2e test, otherwise I guess the e2e test we added will fail.

**Comment by changlong-liu** [comment]
> @changlong-liu Changlong Liu FTE Cam we add the not-generate check option in the e2e test, otherwise I guess the e2e test we added will fail.

Hi @qiaozha , added the e2e PR: https://github.com/amecodegenbot/azure-rest-api-specs/pull/6

**Review by qiaozha — APPROVED** [review]

**Pull request merged**